lib/components/joystick/joystick_events.dart

```diff
@@ -0,0 +1,45 @@
+enum JoystickMoveDirectional {
+  MOVE_UP,
+  MOVE_UP_LEFT,
+  MOVE_UP_RIGHT,
+  MOVE_RIGHT,
+  MOVE_DOWN,
+  MOVE_DOWN_RIGHT,
+  MOVE_DOWN_LEFT,
+  MOVE_LEFT,
+  IDLE
+}
+
+enum ActionEvent { DOWN, UP, MOVE, CANCEL }
+
+class JoystickDirectionalEvent {
```

Comment:
please disregard if this doesn't make sense! but should this event have an id as well, in case the user wants two directional sticks? some games have that

Reply:
Left directional on this component, only one is supported. But you can add several `JoystickAction` with directional enabled.